### PR TITLE
fix(includers/openapi): default value in query

### PR DESCRIPTION
**Pull request by amosov-f**
fix undefiened & default values

Снимок экрана 2023-01-26 в 16 42 32

### CODE REVIEW

File: src/services/includers/batteries/openapi/generators/endpoint.ts [hunk 1]
```diff
@@ -84,12 +84,15 @@ function parameters(params?: Parameters) {
...
     const row = prepareTableRowData({}, param.name, param.schema);
-    let description = param.description;
+    let description = param.description ?? '';
     if (row.description.length) {
-        description += `<br>${row.description}`;
+        description = concatNewLine(description, row.description);
     }
-    if (param.example) {
-        description += `<br>Example: \`${param.example}\``;
+    if (param.example !== undefined) {
```
Comment (moki) [inline src/services/includers/batteries/openapi/generators/endpoint.ts:91]: why not `param.example != null` why should we render:

```
`Example: \`${null}\``
```

File: src/services/includers/batteries/openapi/generators/endpoint.ts [hunk 2]
```diff
@@ -84,12 +84,15 @@ function parameters(params?: Parameters) {
...
+        description = concatNewLine(description, `Example: \`${param.example}\``);
+    }
+    if (param.default !== undefined) {
```
Comment (moki) [inline src/services/includers/batteries/openapi/generators/endpoint.ts:94]: why not `param.default != null` why should we render:

```
`Example: \`${null}\``
```

File: src/services/includers/batteries/openapi/types.ts [hunk 3]
```diff
@@ -100,8 +100,9 @@ export type Parameter = {
     name: string;
     in: In;
     required: boolean;
-    description: string;
-    example: string | number;
+    description?: string;
+    example?: string | number | boolean;
```
Comment (moki) [inline src/services/includers/batteries/openapi/types.ts:104]: would love separate type for that, that will signal about what kind of data we support

File: src/services/includers/batteries/openapi/types.ts [hunk 4]
```diff
@@ -100,8 +100,9 @@ export type Parameter = {
...
+    default?: string | number | boolean;
```
Comment (moki) [inline src/services/includers/batteries/openapi/types.ts:105]: same here

**Review by moki — CHANGES_REQUESTED** [review]

File: src/services/includers/batteries/openapi/generators/traverse.ts [hunk 5]
```diff
@@ -197,6 +197,6 @@ function merge(value: JSONSchema6 | boolean): JSONSchema6 | undefined {
     return value;
 }
 
-function concatNewLine(prefix: string, suffix: string) {
+export function concatNewLine(prefix: string, suffix: string) {
```
Comment (moki) [inline src/services/includers/batteries/openapi/generators/traverse.ts:200]: if its used somewhere else factor it out to `common` module

File: src/services/includers/batteries/openapi/generators/endpoint.ts (hunk 1 above)
Comment (amosov-f) [reply src/services/includers/batteries/openapi/generators/endpoint.ts:91]: null is valid default value

File: src/services/includers/batteries/openapi/generators/endpoint.ts (hunk 2 above)
Comment (amosov-f) [reply src/services/includers/batteries/openapi/generators/endpoint.ts:94]: null is valid default value

File: src/services/includers/batteries/openapi/types.ts (hunk 3 above)
Comment (amosov-f) [reply src/services/includers/batteries/openapi/types.ts]: fixed

File: src/services/includers/batteries/openapi/types.ts (hunk 4 above)
Comment (amosov-f) [reply src/services/includers/batteries/openapi/types.ts]: fixed

File: src/services/includers/batteries/openapi/generators/traverse.ts (hunk 5 above)
Comment (amosov-f) [reply src/services/includers/batteries/openapi/generators/traverse.ts]: fixed

**Review by 3y3 — APPROVED** [review]

**Review by moki — APPROVED** [review]

**Pull request merged**